src/rsch/nrf_802154_rsch.c

```
@@ -570,3 +572,13 @@ void nrf_802154_wifi_coex_denied(void)
     prec_approved_prio_set(RSCH_PREC_COEX, RSCH_PRIO_RX);
     notify_core();
 }
+
+uint32_t nrf_802154_rsch_get_next_scheduled_receive_time(void)
+{
+    return m_dly_ts[RSCH_DLY_RX].t0 + m_dly_ts[RSCH_DLY_RX].dt;
```

Note that neither this function's implementation nor its callers check if the provided time represents an actually scheduled, pending delayed RX window. Could you have a look into the 802.15.4-2015 spec to determine what should be done with the CSL phase when this time is in the past or happens to be invalid for any other reason (e.g. the delayed timeslot was cancelled)? Currently it silently passes the possibly invalid data to the frame, the spec might have different requirements.